google/cloud/bigquery/query.py

```diff
@@ -329,18 +333,41 @@ class ScalarQueryParameter(_AbstractQueryParameter):
             Parameter name, used via ``@foo`` syntax.  If None, the
             parameter can only be addressed via position (``?``).
 
-        type_ (str):
-            Name of parameter type.  One of 'STRING', 'INT64',
-            'FLOAT64', 'NUMERIC', 'BIGNUMERIC', 'BOOL', 'TIMESTAMP', 'DATETIME', or
-            'DATE'.
+        type_ (Union[str, google.cloud.bigquery.query.ScalarQueryParameterType]):
+            Name of parameter type. See
+            :class:`google.cloud.bigquery.enums.SqlTypeNames` and
+            :class:`google.cloud.bigquery.enums.SqlParameterScalarTypes` for
+            supported types.
 
-        value (Union[str, int, float, decimal.Decimal, bool, datetime.datetime, datetime.date]):
+        value (Union[ \
+            str, int, float, dateutil.relativedelta.relativedelta, \
+            decimal.Decimal, bool, datetime.datetime, datetime.date \
+        ]):
             The scalar parameter value.
     """
 
-    def __init__(self, name, type_, value):
+    def __init__(
+        self,
+        name: Optional[str],
+        type_: Optional[Union[str, ScalarQueryParameterType]],
+        value: Optional[
+            Union[
+                str,
+                int,
+                float,
+                dateutil.relativedelta.relativedelta,
+                decimal.Decimal,
+                bool,
+                datetime.datetime,
+                datetime.date,
+            ]
+        ],
+    ):
         self.name = name
-        self.type_ = type_
+        if isinstance(type_, ScalarQueryParameterType):
```

Comment:
+1 for accepting `ScalarQueryParameterType`

Reply:
Thanks. I think I'll split this out into its own PR.

Reply:
Mailed https://github.com/googleapis/python-bigquery/pull/850 with just this change since it's unrelated to INTERVAL (which actually doesn't support query parameters yet)